packages/caliper-core/test/worker/caliper-worker.js

```diff
@@ -161,5 +190,101 @@ describe('Caliper worker', () => {
             await worker.executeRound(mockTestMessage);
             validateCallsAndWarnings(4);
         });
+
+        [5, 10].forEach(numberOfTxs => {
+            it(`should run ${numberOfTxs} transactions and wait for completion when no errors occur`, async () => {
+                const worker = new CaliperWorker(mockConnector, 1, mockMessenger, 'uuid');
+                await worker.prepareTest(mockTestMessage);
+
+                mockTestMessage.getNumberOfTxs.returns(numberOfTxs);
+                mockTestMessage.getRoundDuration.returns(null);
+
+                mockWorkload.submitTransaction.resetHistory();
+                mockStats.getTotalSubmittedTx.resetHistory();
+                mockStats.getTotalFinishedTx.resetHistory();
+                mockStats.getCumulativeTxStatistics.resetHistory();
+
+                let submittedTx = 0;
+                let finishedTx = 0;
+
+                // Stub the methods
+                mockStats.getTotalSubmittedTx.callsFake(() => submittedTx);
+                mockStats.getTotalFinishedTx.callsFake(() => finishedTx);
+                mockStats.getCumulativeTxStatistics.returns({});
+
+                worker.internalTxObserver.getCurrentStatistics = () => mockStats;
+
+                mockWorkload.submitTransaction.callsFake(async () => {
+                    submittedTx += 1;
+                    finishedTx += 1;
+                    return Promise.resolve();
+                });
+
+                await worker.executeRound(mockTestMessage);
+
+                sinon.assert.callCount(mockWorkload.submitTransaction, numberOfTxs);
+                sinon.assert.calledOnce(deactivateMethod);
+                sinon.assert.calledOnce(mockRate.end);
+                sinon.assert.calledOnce(mockWorkload.cleanupWorkloadModule);
+                sinon.assert.called(mockConnector.releaseContext);
+            });
+        });
+
+        it('should execute the round for a specified duration', async function() {
+            this.timeout(5000); // Increase the timeout for this test
+            const worker = new CaliperWorker(mockConnector, 1, mockMessenger, 'uuid');
+            await worker.prepareTest(mockTestMessage);
+            mockWorkload.submitTransaction.resolves();
+            mockTestMessage.getRoundDuration.returns(1); // duration in seconds
+
+            await worker.executeRound(mockTestMessage);
+
+            sinon.assert.calledOnce(deactivateMethod);
+            sinon.assert.calledOnce(mockRate.end);
+            sinon.assert.calledOnce(mockWorkload.cleanupWorkloadModule);
+            sinon.assert.called(mockConnector.releaseContext);
+        });
+
+
+        it('should handle errors during the prepareTest phase', async () => {
```

Comment:
We need more tests for when a prepare message is received

Reply:
I added some more test specifically testing around the prepare message makeup, is this what you meant?

Reply:
I think it's covered here now, although the it strings could be a little more descriptive I think you have covered what is needed.